### PR TITLE
[MU4] Preferences. Folders

### PR DESCRIPTION
**Pull request by Eism**
image

### CODE REVIEW

File: src/appshell/view/folderspreferencesmodel.cpp [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+void FoldersPreferencesModel::load()
+{
+    beginResetModel();
+
+    m_folders = {
+        { FolderType::Scores, qtrc("appshell", "Scores"), scoresPath() },
+        { FolderType::Styles, qtrc("appshell", "Styles"), stylesPath() },
+        { FolderType::Templates, qtrc("appshell", "Templates"), templatesPath() },
+        { FolderType::Plugins, qtrc("appshell", "Plugins"), pluginsPath() },
+        { FolderType::SoundFonts, qtrc("appshell", "Soundfonts"), "" }, // todo: need implement
```
Comment (RomanPudashkin) [inline src/appshell/view/folderspreferencesmodel.cpp:85]: Sound**F**onts?

File: src/extensions/internal/extensionsconfiguration.cpp [hunk 2]
```diff
@@ -246,7 +245,30 @@ io::paths ExtensionsConfiguration::templatesPaths() const
...
 }
 
+ValCh<io::path> ExtensionsConfiguration::extensionsPath() const
+{
+    ValCh<io::path> result;
+    result.ch = m_extensionsPathChanged;
+
+    result.val = settings()->value(USER_EXTENSIONS_PATH).toString();
+    if (result.val.empty()) {
+        result.val = settings()->defaultValue(USER_EXTENSIONS_PATH).toString();
```
Comment (RomanPudashkin) [inline src/extensions/internal/extensionsconfiguration.cpp:255]: You can call setDefaultValue in ExntesionsConfiguration::init and use Settings::value directly:

 ValCh<io::path> ExtensionsConfiguration::extensionsPath() const
 {
 ValCh<io::path> result;
 result.ch = m_extensionsPathChanged;
 result.val = settings()->value(USER_EXTENSIONS_PATH);
 
 return result;
 }

Comment (Eism) [reply src/extensions/internal/extensionsconfiguration.cpp]: thanks, fixed

File: src/appshell/view/folderspreferencesmodel.cpp (hunk 1 above)
Comment (Eism) [reply src/appshell/view/folderspreferencesmodel.cpp]: thanks, fixed

**Review by RomanPudashkin — APPROVED** [review]

**Pull request merged**